src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -204,754 +176,13 @@ private void initiate()
-    {
-        //method returns false if there is no next block (structures finished)
-        if (!job.getStructure().findNextBlockSolid())
-        {
-            job.getStructure().reset();
-            incrementBlock();
-            return AIState.BUILDER_DECORATION_STEP;
-        }
-        return this.getState();
-    }
-
-    private AIState findNextBlockNonSolid()
-    {
-        //method returns false if there is no next block (structures finished)
-        if (!job.getStructure().findNextBlockNonSolid())
-        {
-            job.getStructure().reset();
-            incrementBlock();
-            return AIState.BUILDER_COMPLETE_BUILD;
-        }
-        return this.getState();
-    }
-
-    @NotNull
-    private AIState completeBuild()
-    {
-        if (job.getStructure() == null)
-        {
-            //fix for bad structures
-            job.complete();
-        }
-        job.getStructure().getEntities().forEach(this::spawnEntity);
-
-        final String structureName = job.getStructure().getName();
-
-        LanguageHandler.sendPlayersLocalizedMessage(worker.getColony().getMessageEntityPlayers(),
-          "entity.builder.messageBuildComplete",
-          structureName);
-
-        final WorkOrderBuild wo = job.getWorkOrder();
-        if (wo == null)
-        {
-            Log.getLogger().error(String.format("Builder (%d:%d) ERROR - Finished, but missing work order(%d)",
-              worker.getColony().getID(),
-              worker.getCitizenData().getId(),
-              job.getWorkOrderId()));
-        }
-        else
-        {
-            if (!(wo instanceof WorkOrderBuildDecoration))
-            {
-                final AbstractBuilding building = job.getColony().getBuilding(wo.getBuildingLocation());
-                if (building == null)
-                {
-                    Log.getLogger().error(String.format("Builder (%d:%d) ERROR - Finished, but missing building(%s)",
-                      worker.getColony().getID(),
-                      worker.getCitizenData().getId(),
-                      wo.getBuildingLocation()));
-                }
-                else
-                {
-                    building.setBuildingLevel(wo.getUpgradeLevel());
-                }
-            }
-            else
-            {
-                if (structureName.contains(WAYPOINT_STRING))
-                {
-                    worker.getColony().addWayPoint(wo.getBuildingLocation(), world.getBlockState(wo.getBuildingLocation()));
-                }
-            }
-            job.complete();
-        }
-
-        final AbstractBuilding workerBuilding = getOwnBuilding();
-        if (workerBuilding instanceof BuildingBuilder)
-        {
-            ((BuildingBuilder) workerBuilding).resetNeededResources();
-        }
-        resetTask();
-        worker.addExperience(XP_EACH_BUILDING);
-        workFrom = null;
-
-        return AIState.IDLE;
+        return AIState.START_BUILDING;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Unnecessary use of fully qualified name 'AIState.START_BUILDING' due to existing static import 'com.minecolonies.coremod.entity.ai.util.AIState.*' [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AUnnecessaryFullyQualifiedName)